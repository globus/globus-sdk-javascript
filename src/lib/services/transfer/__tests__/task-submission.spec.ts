import serviceTestSuite from '../../../../__utils__/service-test-suite';
import { taskSubmission } from '..';

serviceTestSuite('transfer', 'task-submission', (fetch) => {
  test('submissionId', async () => {
    await taskSubmission.submissionId();
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(
      `https://transfer.api.globusonline.org/v0.10/submission_id`,
      {
        headers: {},
      },
    );
  });

  test('submitDelete', async () => {
    const payload = {
      endpoint: 'e434fe90-591f-40d9-8d34-3016b6237046',
      submission_id: '65d1dc8c-5ebb-487e-8990-e5a25e3cf1d4',
      DATA: [{ DATA_TYPE: 'delete_item', path: '/' }],
    };

    await taskSubmission.submitDelete({
      payload,
    });
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(
      `https://transfer.api.globusonline.org/v0.10/delete`,
      expect.objectContaining({
        body: expect.stringContaining(`"DATA_TYPE":"delete"`) as unknown,
        headers: expect.objectContaining({
          'Content-Type': 'application/json',
        }) as unknown,
        method: 'POST',
      }),
    );
  });

  test('submitTransfer', async () => {
    await taskSubmission.submitTransfer({
      payload: {},
    });
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(
      `https://transfer.api.globusonline.org/v0.10/transfer`,
      expect.objectContaining({
        body: expect.stringContaining(`"DATA_TYPE":"transfer"`) as unknown,
        headers: expect.objectContaining({
          'Content-Type': 'application/json',
        }) as unknown,
        method: 'POST',
      }),
    );
  });
});
