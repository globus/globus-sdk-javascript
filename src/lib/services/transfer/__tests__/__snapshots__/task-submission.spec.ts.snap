// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transfer.task-submission submissionId 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "method": "GET",
  "url": "https://transfer.api.globusonline.org/v0.10/submission_id",
}
`;

exports[`transfer.task-submission submitDelete 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "content-length": "175",
    "content-type": "application/json",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "method": "POST",
  "url": "https://transfer.api.globusonline.org/v0.10/delete",
}
`;

exports[`transfer.task-submission submitTransfer 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "content-length": "24",
    "content-type": "application/json",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "method": "POST",
  "url": "https://transfer.api.globusonline.org/v0.10/transfer",
}
`;
